### PR TITLE
Handle more then 3 peg holes

### PR DESCRIPTION
**Pull request by JosefMeixner**
I added support to use pegbars with more than three holes. Due to the poor availability of paper for traditional pegbars I use a puncher for a filing system with six round holes and a custom built pegbar.

The code assumed the pegbar specification to have exactly three holes. The auto center code didn't work when a peg hole was at the top border of the paper (I scanned a bit poorly) and when there are more then three holes the original code centered the page at the middle hole of the three best matching holes.

I think I could also get the code to work with only two peg holes, but have not tried yet. It might help people try out toonz (a pegbar can be easily created with a 3d-printer), but it would be also quite hard to draw and keep the pages aligned, so don't know if it is worth the try.

### CODE REVIEW

**Comment by kogaki** [comment]
Jenkins, confirm this

File: toonz/sources/toonzlib/autopos.cpp [hunk 1]
```diff
@@ -1576,53 +1599,49 @@ int get_image_rotation_and_center(const TRasterP &img, int strip_width,
...
-static int compare_dots(DOT dots[], int *ndots,
-						DOT reference[], int ref_dot)
+static int compare_dots(DOT const dots[], int ndots, DOT reference[], int ref_dot, int& i_ok, int& j_ok, int& k_ok)
 {
 	int found;
-	int toll, i_ok = 0, j_ok = 0, k_ok = 0;
+	int toll;
 	float tolld;
 	int i, j, k;
 	short *dot_ok = 0;
```
Comment (kogaki) [inline toonz/sources/toonzlib/autopos.cpp]: It looks dot_ok should be `bool`

**Comment by JosefMeixner** [comment]
Thank you, I missed that.

**Comment by kogaki** [comment]
Jenkins, confirm this

**Comment by kogaki** [comment]
OK. LGTM.

**Pull request merged**